lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/diagnostics/PluginValidator.java

```diff
@@ -29,24 +30,29 @@
 import org.eclipse.lemminx.extensions.maven.utils.MavenPluginUtils;
 import org.eclipse.lsp4j.Diagnostic;
 import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
 
 class PluginValidator {
 
 	private static final Logger LOGGER = Logger.getLogger(PluginValidator.class.getName());
 
 	private MavenLemminxExtension plugin;
-
-	public PluginValidator(MavenLemminxExtension plugin) {
+	private CancelChecker cancelChecker;
+	
+	public PluginValidator(MavenLemminxExtension plugin, CancelChecker cancelChecker) {
 		this.plugin = plugin;
+		this.cancelChecker = cancelChecker;
 	}
 
-	public Optional<List<Diagnostic>> validatePluginResolution(DiagnosticRequest diagnosticRequest) {
+	public Optional<List<Diagnostic>> validatePluginResolution(DiagnosticRequest diagnosticRequest) throws CancellationException {
+		cancelChecker.checkCanceled();
 		try {
 			MavenPluginUtils.getContainingPluginDescriptor(diagnosticRequest.getNode(), plugin, true);
 		} catch (PluginResolutionException | PluginDescriptorParsingException | InvalidPluginDescriptorException e) {
 			LOGGER.log(Level.WARNING, "Could not resolve plugin description", e);
 
 			// Add artifactId diagnostic
+			cancelChecker.checkCanceled();
 			String errorMessage = e.getMessage();
 			DOMNode pluginNode = DOMUtils.findClosestParentNode(diagnosticRequest.getNode(), "plugin");
 			Optional<DOMNode> artifactNode = pluginNode.getChildren().stream().filter(node -> !node.isComment())
```

Comment:
Do you think it could have some DOMUtils methods which could take time, if yes we should use cancelChecker.

Reply:
The majority of `DOMUtil` methods are the "short" ones, working with DOM document nodes (like finding children, parents or gathering the text from node value).  So for the most of them the answer is - no, they're pretty quick.
However it worth double-checking the class for possible "long-runners".